tuning/libtuner.py

```diff
@@ -184,113 +194,63 @@ class ParsedDisptachBenchmarkResult:
+        time_str = self.extract_key(pattern)
+        if not time_str:
             return None
         try:
-            return float(self.get_tokens()[3])
+            return float(time_str)
         except ValueError:
             return None
 
-    def generate_sample_result(
-        self, candidate_id: int = 0, mean_time: float = random.uniform(100.0, 500.0)
-    ) -> str:
-        # time unit is implicit and dependent on the output of iree-benchmark-module
-        return f"{candidate_id}\tMean Time: {mean_time:.1f}\n"
-
-
-@dataclass
-class ModelBenchmarkResult:
-    result_str: Optional[str] = None
-
-    def get_tokens(self) -> list[str]:
-        # e.g. ['Benchmarking:', '/sdxl-scripts/tuning/tuning_2024_07_19_08_55/unet_candidate_12.vmfb', 'on', 'device', '4', 'BM_main/process_time/real_time_median', '65.3', 'ms', '66.7', 'ms', '5', 'items_per_second=15.3201/s']
-        if self.result_str is None:
-            return []
-        try:
-            return self.result_str.split()
-        except:
-            return []
-
-    def get_model_candidate_path(self) -> Optional[str]:
-        if len(self.get_tokens()) < 2:
-            return None
-        return self.get_tokens()[1]
-
-    def get_candidate_id(self) -> Optional[int]:
-        if self.get_model_candidate_path():
-            try:
-                path_str = self.get_model_candidate_path()
-                return int(path_str.split("_")[-1].split(".")[0]) if path_str else None
-            except ValueError:
-                return None
-        return None
-
-    def get_device_id(self) -> Optional[int]:
-        if len(self.get_tokens()) < 5:
-            return None
-        try:
-            return int(self.get_tokens()[4])
-        except ValueError:
-            return None
-
-    def get_benchmark_time(self) -> Optional[int | float]:
-        if len(self.get_tokens()) < 7:
+    def get_median_time(self) -> Optional[float]:
+        pattern = r"process_time/real_time_median\s+([\d.]+)\s\w{2}"
+        time_str = self.extract_key(pattern)
+        if not time_str:
             return None
         try:
-            return float(self.get_tokens()[6])
+            return float(time_str)
         except ValueError:
             return None
 
-    def get_calibrated_result_str(self, change: float) -> Optional[str]:
-        if self.result_str is None:
-            return self.result_str
-
-        benchmark_time = self.get_benchmark_time()
-        if benchmark_time is None:
-            return self.result_str
 
-        # Format the change to be added to the string
-        percentage_change = change * 100
+def generate_display_DBR(
+    candidate_id: int = 0, mean_time: float = random.uniform(100.0, 500.0)
+) -> str:
+    """Generate dispatch_benchmark_result string for displaying"""
+    # time unit is implicit and dependent on the output of iree-benchmark-module
+    return f"{candidate_id}\tMean Time: {mean_time:.1f}\n"
```

Comment:
This is no longer true when we switch away from the shell script -- we can see the time unit in the direct output of iree benchmark module.